### ISSUE TITLE
Timing Issue Offsync

### ISSUE THREAD

**Issue by BurningFlames2314** [ppy/osu#18652]
### Type

Game behaviour

### Bug description

When playing certain beatmaps, there is a offsync between the song and hitnotes... Not sure about how many songs this affects but the one I found is this ranked map.

https://osu.ppy.sh/beatmapsets/440650#osu/1573659

In the regular osu! the timing is fine but in lazer for some reason the hit notes slowly gets more off beat the longer I play it (I've check both in edit mode and playing mode)

### Version

2022.612.0-lazer

### Logs

[legacy-ipc.log](https://github.com/ppy/osu/files/8885507/legacy-ipc.log)
[network.log](https://github.com/ppy/osu/files/8885508/network.log)
[performance.log](https://github.com/ppy/osu/files/8885509/performance.log)
[runtime.log](https://github.com/ppy/osu/files/8885510/runtime.log)
[updater.log](https://github.com/ppy/osu/files/8885511/updater.log)
[database.log](https://github.com/ppy/osu/files/8885512/database.log)

**Comment by BurningFlames2314** [comment]
edit:
edit mode timing when you skip parts of the song becomes f***ed up...
playing from start to finish doesnt seem to cause this bug. but when pausing during breaks  (for some period of time, i usually pause for a couple of minutes in breaks to check messages etc.) it causes the rest of the map to desync just like in edit mode.

**Comment by frenzibyte** [comment]
Can't seem to reproduce on macOS, may be a platform-specific/Windows-only shortcoming caused by https://github.com/ppy/osu-framework/pull/5238.

**Comment by peppy** [comment]
I haven't seen this happen on windows, so potentially unrelated.

@BurningFlames2314 Could you please install [the previous release](https://github.com/ppy/osu/releases/tag/2022.605.1) and see if you can reproduce? Note that it will auto-update to the newest release when you restart it, so you will need to run the installer each time to revert back.

**Comment by BurningFlames2314** [comment]
up let me try that, ill also record a video of it just in case... give a bit of time

**Comment by peppy** [comment]
Also to rule out any doubt, after trying this please also update back to the latest and confirm it happens there again, using the same testing flow.

**Comment by BurningFlames2314** [comment]
I have updated back to latest driver and the bug still persist...

Current release video: 
https://youtu.be/h4kgcKF8Mfg
`<iframe width="560" height="315" src="https://www.youtube.com/embed/h4kgcKF8Mfg" title="YouTube video player" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture" allowfullscreen></iframe>`

Previous release video: Does not produce the bug
https://youtu.be/Kv-lmRCZFZk
`<iframe width="560" height="315" src="https://www.youtube.com/embed/Kv-lmRCZFZk" title="YouTube video player" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture" allowfullscreen></iframe>`

never tried embedding videos on this... hopefully im doing it right lol

**Comment by BurningFlames2314** [comment]
nvm xD

**Comment by peppy** [comment]
Seems like a potential regression then. Interesting that no one else has reported issues yet. I'll try to repro on native windows later today.

@BurningFlames2314 what kind of HDD/SSD are you running on?

**Comment by Deezelchonk** [comment]
I wanted to report with an issue that makes osu!lazer on mobile ALMOST impossible to get a clear on a map but since I don’t have pc, I can’t report it (I only have a phone)

**Comment by peppy** [comment]
I think you may be in the wrong place, unless you're referring specifically to a regression in the latest release only.

**Comment by BurningFlames2314** [comment]
SSD = ADATA SX8200PNP
HDD = WDC WD40EZAZ-00SF3B0

**Comment by MaxKruse** [comment]
As asked in #18659 i tried reproducing this.

Quick Testing methodology:
1. Play map, pause after "lag" spike, then continue play
2. Play map, then pause, tab out, and tab back in to play
3. Open the editor and skip around in the timeline a few times (5-6 times is enough on my machine)

Here are my results:

 - 605.1: I **can not** reproduce the audio issue, whatsoever
 - 612.0:  Reproduced this bug in the following cases:
 1. Editor timeline skipping reproduces this bug
 2. Play the map, pause after "lag" spike, then continue play reproduces this bug

Pausing and tabbing in and out does not seem to cause this issue.

My very naive guess is that samples are internally being tracked as being played, when they were unloaded by some mechanism before. From my very short testing, it seems that the music playback is ahead what the game *thinks*, by about 70-90ms in my test runs (assuming having to tap basically at the start of hitobject's hitwindow being active to achieve the desired effect of hitsounds being played to the music, as opposed to later).

**Comment by MaxKruse** [comment]
The audio device in my case is a FiiO USB DAC-E10

**Comment by BurningFlames2314** [comment]
Mine is the Scarlett Solo USB

**Comment by 25pwn** [comment]
I'm getting the same issue on Windows, I'll test linux later. Happens after breaks, pauses and replay rewinds are affected too.

**Comment by 25pwn** [comment]
Can confirm it's a threading issue because turning off multithreading resolves the issue. Doesn't happen on Linux(Wayland+Pipewire).

**Comment by 25pwn** [comment]
Can confirm it's a threading issue because turning off multithreading prevents this from happening(unless a lag spike occurs). Doesn't happen on Linux(Wayland+Pipewire).

**Comment by peppy** [comment]
> Can confirm it's a threading issue because turning off multithreading prevents this from happening(unless a lag spike occurs). Doesn't happen on Linux(Wayland+Pipewire).

Can you also confirm that the previous release fixes the issue for you on both threading modes? I'm still trying to understand what the causation of the issue is so would appreciate if each user chiming in on this thread can confirm this.

Also can everyone confirm whether they can feel the issue when playing with hitsounds disabled? This will confirm whether it is an issue with playback of samples, or whether the fundamentals of audio clock timing are off.

I'm also going to prepare a release without the one change we made to bass configuration and will post here for further testing.

**Comment by peppy** [comment]
For anyone experiencing this issue, could you please try [this release](https://github.com/ppy/osu/releases/tag/no-async-bass-test) and see if you can still reproduce the issue?

Once done testing, you can revert to a real release by running another install over the top.

**Comment by BurningFlames2314** [comment]
tried out the test release, initial observations looks good, the bug doesnt seem to appear anymore, Ill get back to you in a bit and try a few more things out.

**Comment by peppy** [comment]
I'd also appreciate testing on the broken release with effect volume zero.

**Comment by BurningFlames2314** [comment]
yep, im working on it, ill give the details in a bit

**Comment by BurningFlames2314** [comment]
Current release update:
Did 4 attempts on https://osu.ppy.sh/beatmapsets/1773106#osu/3630306 Not classic Extra dif
2 attempts with pause (1 with hitsound and one without)
Both attempts shows significant increase in 100s and hitcircles gets delayed average intial UR before pause 120 and after UR 180
2 attempts without pause (same with above)
Average UR 110

TLDR hitcircles are the one getting delayed, not the hitsounds

**Comment by peppy** [comment]
Thanks a lot. We're going to push out a hotfix for this (and this thread will be closed) but please keep a watch on this thread as I may require more diagnostics in the future, since I cannot reproduce this bug and we will want to report it to the developer that maintains the audio library we use.

**Comment by BurningFlames2314** [comment]
should i mark it as closed then?

**Comment by peppy** [comment]
@Wieku if you happen to beat us in reporting it to Un4seen, please link the report thread here. It's on my todo list but also quite low priority since we have a workaround for now.
